src/bridge/bridge.c

```diff
@@ -177,7 +177,8 @@ static void
 setup_dbus_daemon (gpointer addrfd)
```

Comment:
This is begging for a rewrite: hardcode `--print-address=3` and `dup2()` the fd over, and just close everything from 4 and up.

Even better: use `GSubproces` instead, with `g_subprocess_launcher_take_fd()`, and drop the child setup function.

Reply:
Done. Tested with:
```
❱❱❱ make && env -u DBUS_SESSION_BUS_ADDRESS G_MESSAGES_DEBUG=all ./cockpit-bridge | cat
[...]
(cockpit-bridge:232955): cockpit-bridge-DEBUG: 08:49:54.609: launched dbus-daemon: 232957
(cockpit-bridge:232955): cockpit-bridge-DEBUG: 08:49:54.613: session bus address: unix:abstract=/tmp/dbus-uolitsRRHL,guid=1fe30868a2766df409b2654961024f92
```
... and I can see the process properly in `ps`. This results in 31 fewer lines of code, and looks quite nice indeed. Thanks for pointing out, I didn't know that API yet.